src/librustc_metadata/rmeta/decoder.rs

```diff
@@ -1723,9 +1811,6 @@ impl CrateMetadata {
         private_dep: bool,
         host_hash: Option<Svh>,
     ) -> CrateMetadata {
-        let def_path_table = record_time(&sess.perf_stats.decode_def_path_tables_time, || {
-            root.def_path_table.decode((&blob, sess))
```

Comment:
`def_path_table` didn't actually needed the session for decoding.
So `def_key` and `def_path_hash` decoding functions don't need the session argument either.
If it's removed the patch will become noticeably cleaner.